src/components/Dao/DaoHistoryPage.tsx

```diff
@@ -129,41 +131,113 @@ export default withSubscription({
             closingAt_lte: "${Math.floor(new Date().getTime() / 1000)}"
           }
         ){
-          ...ProposalFields
+          id
+          accountsWithUnclaimedRewards
+          boostedAt
+          closingAt
+          confidenceThreshold
+          createdAt
+          dao {
+            id
+            schemes {
+              id
+              address
+            }
+          }
+          description
+          descriptionHash
+          executedAt
+          executionState
+          expiresInQueueAt
+          genesisProtocolParams {
+            id
+            activationTime
+            boostedVotePeriodLimit
+            daoBountyConst
+            limitExponentValue
+            minimumDaoBounty
+            preBoostedVotePeriodLimit
+            proposingRepReward
+            queuedVotePeriodLimit
+            queuedVoteRequiredPercentage
+            quietEndingPeriod
+            thresholdConst
+            votersReputationLossRatio
+          }
+          gpRewards {
+            id
+          }
+          scheme {
+            ...PluginFields
+          }
+          gpQueue {
+            id
+            threshold
+            votingMachine
+          }
+          organizationId
+          preBoostedAt
+          proposer
+          quietEndingPeriodBeganAt
+          stage
+          stakesFor
+          stakesAgainst
+          tags {
+            id
+          }
+          totalRepWhenCreated
+          totalRepWhenExecuted
+          title
+          url
+          votesAgainst
+          votesFor
+          votingMachine
+          winningOutcome
```

Comment:
This whole thing is nasty.  Is it all needed?  No way to automate it, like reflection?  Can we please add some code comments to describe exactly what is going on and why?

Reply:
The reason this is needed now is described here: https://github.com/daostack/arc.js/issues/471

Reply:
I'll point to this in a comment. In the future this can all be removed, and the nested where filters can be passed into the normal search/state function.